src/sentry/workflow_engine/types.py

```diff
@@ -32,17 +33,19 @@ class DetectorPriorityLevel(IntEnum):
 ProcessedDataConditionResult = tuple[bool, list[DataConditionResult]]
 
 
-class EventJob(TypedDict):
+@dataclass(frozen=True)
+class EventJob:
     event: GroupEvent
```

Comment:
Could `event` just be an attribute of `WorkflowJob` rather than using OOP to do this inheritance?  (this breaks the inversion principle in solid programming)